slider/beatmap.py

```diff
@@ -31,6 +31,145 @@ def _get(cs, ix, default=no_default):
         return default
 
 
+class Event:
+
+    EventType = Enum('EventType',
+                     {'Background': 0,
+                      'Video': 1,
+                      'Break': 2,
+                      'Sprite': 3,
+                      'Animation': 4})
+
+    @property
+    def start_time(self):
+        return self.start_time if self.start_time != 0 else None
+
+    @classmethod
+    def parse(cls, data):
+        event_type, start_time, *event_params = data.split(',')
+        try:
+            start_time = int(start_time)
+            start_time = timedelta(milliseconds=start_time)
+        except ValueError:
+            return Storyboard.parse()
+        try:
+            event_type = cls.EventType(int(event_type))
+        except ValueError:
+            try:
+                event_type = cls.EventType[event_type]
+            except KeyError:
+                raise ValueError(f'Invalid event type, got {event_type}')
+        if event_type == cls.EventType.Background:
+            pass
+        elif event_type == cls.EventType.Video:
+            pass
+        elif event_type == cls.EventType.Break:
+            return Break.parse(start_time, event_params)
+        # should be unreachable, added to enforce explicit handling of enums.
+        else:
+            raise ValueError(
+                f'Unimplemented event type: {event_type}')
+
+
+class Background(Event):
+
+    @property
+    def start_time(self):
+        return self.start_time
+
+    @property
+    def event_type(self):
+        return self.EventType.Background
+
+    @classmethod
+    def parse(cls, start_time, event_params):
+        try:
+            filename, x_offset, y_offset = event_params
+        except ValueError:
+            raise ValueError(
+                f'Missing param for Background, received {event_params}')
+        try:
+            x_offset = int(x_offset)
+        except ValueError:
+            raise ValueError(f'x_offset is invalid, got {x_offset}')
+        try:
+            y_offset = int(y_offset)
+        except ValueError:
+            raise ValueError(f'y_offset is invalid, got {y_offset}')
+
```

Comment:
missing `return cls(...)`?

Reply:
lol as you can see I didn't test this :cry: